File: Python/14_Longest_Common_Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        for x in range(len(strs[0])):
            for y in range(1,len(strs)):
                if (x < len(strs[y]) and strs[y-1][x] == strs[y][x]):
                    continue 
                else:
                    return result
            result += strs[0][x]
        return result

File: Java/14_Longest_Common_Prefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = "";
        int counter = 0;
        for (int i = 0; i < strs[0].length(); i++) {
            for (int j = 1; j < strs.length; j++) {
                if ((counter >= strs[j].length()) || strs[0].charAt(counter)!= strs[j].charAt(counter)) {
                    return prefix;
                }
            }
            prefix += strs[0].charAt(counter);
            counter++;
        }
        
        return prefix;
    }
}